dl/src/main/scala/com/intel/analytics/bigdl/dataset/text/SampleToBatchPadding.scala

```diff
@@ -0,0 +1,167 @@
+        "SampleToBatchPadding: Only Float/Double supported")
+    }
+  }
+
+  override def apply(prev: Iterator[Sample[T]]): Iterator[MiniBatch[T]] = {
+    new Iterator[MiniBatch[T]] {
+      private val featureTensor: Tensor[T] = Tensor[T]()
+      private val labelTensor: Tensor[T] = Tensor[T]()
+      private var featureData: Array[T] = null
+      private var labelData: Array[T] = null
+      private var sampleData: Array[Sample[T]] = null
+
+      private val batchSize = Utils.getBatchSize(totalBatch)
+      private var featureSize: Array[Int] = null
+      private var labelSize: Array[Int] = null
+      private var oneFeatureLength: Int = 0
+      private var oneLabelLength: Int = 0
+      private var featureFirstDim: Int = 0
+      private var labelFirstDim: Int = 0
+      override def hasNext: Boolean = prev.hasNext
+
+      override def next(): MiniBatch[T] = {
+        if (prev.hasNext) {
+          var i = 0
+          featureSize = null
+          labelSize = null
+          if (sampleData == null) sampleData = new Array[Sample[T]](batchSize)
+          while (i < batchSize && prev.hasNext) {
+            val sample = prev.next()
+            require(sample.feature().isContiguous() && sample.label().isContiguous(),
+              "SampleToBatchPadding: Only support contiguous tensor, pls use " +
+                "tensor.contiguous() before batching")
+            if (sampleData(i) == null) sampleData(i) = Sample()
+            if ((featureSize != null) && (!padFeature)) {
+              require(util.Arrays.equals(sample.feature().size(), featureSize),
+                "SampleToBatchPadding: all feature size should be same when not padding feature")
+            }
+            if ((labelSize != null) && (!padLabel)) {
+              require(util.Arrays.equals(sample.label().size(), labelSize),
+                "SampleToBatchPadding: all label size should be same when not padding label")
+            }
+            sampleData(i).copy(sample)
+            if ((featureSize == null) || (sample.feature().nElement() > featureSize.product)) {
+              featureFirstDim = sample.feature().size(1)
+              featureSize = sample.feature().size()
+
+            }
+            if ((labelSize == null) || (sample.label().nElement() > labelSize.product)) {
+              labelFirstDim = sample.label().size(1)
+              labelSize = sample.label().size()
+            }
+            i += 1
+          }
+          val batchLength = i
+
+          // init
+          var padLength = if (padFeature) fixLength.getOrElse(featureFirstDim) else featureFirstDim
+          require(padLength >= featureFirstDim, "SampleToBatchPadding: fixLength " +
+            "should not be less than first dimension of feature when padding")
+          oneFeatureLength = padLength * featureSize.product / featureFirstDim
+          featureSize = Array(batchLength, padLength) ++ featureSize.slice(1, featureSize.length)
```

Comment:
reorder these two lines?
featureSize(padDim) = padLength
oneFeatureLength = featureSize.product

The meaning of featureSize might be confused since it indicates the size of one feature at first and points to a batch of input feature later.